### PR TITLE
add unit test for CreatePVC

### PR DESCRIPTION
**Pull request by syamgk**
add unit-test for CreatePVC

### CODE REVIEW

File: pkg/occlient/occlient_test.go [hunk 1]
```diff
@@ -8,9 +8,72 @@ import (
...
+			}
+
+			// Check for validating actions performed
+			if len(fkclientset.Kubernetes.Actions()) != 1 {
+				t.Errorf("expected 1 action in CreatePVC got: %v", fkclientset.RouteClientset.Actions())
+			}
+			// Checks for return values in positive cases
+			if err == nil {
+				createdPVC := fkclientset.Kubernetes.Actions()[0].(ktesting.CreateAction).GetObject().(*corev1.PersistentVolumeClaim)
+				// fmt.Println(createdPVC)
```
Comment (cdrage) [inline pkg/occlient/occlient_test.go]: remove this line

File: pkg/occlient/occlient_test.go [hunk 2]
```diff
@@ -8,9 +8,72 @@ import (
...
+				// created route should be labeled with labels passed to CreatePVC
+				if !reflect.DeepEqual(createdPVC.Labels, tt.labels) {
+					t.Errorf("labels in created route is not matching expected labels, expected: %v, got: %v", tt.labels, createdPVC.Labels)
+				}
+				//  name and size that route is pointg to should match
```
Comment (cdrage) [inline pkg/occlient/occlient_test.go]: pointg -> pointing

File: pkg/occlient/occlient_test.go [hunk 3]
```diff
@@ -8,9 +8,72 @@ import (
...
+				//  if createdPVC.Size != tt.size {
+				//      t.Errorf("route is not matching to expected service name, expected: %s, got %s", tt.size, createdPVC.Size)
+				//  }
```
Comment (cdrage) [inline pkg/occlient/occlient_test.go]: remove these lines of code?

**Comment by cdrage** [comment]
@syamgk You need to add a description to all your commit messages as well! It helps when generating a changelog.

File: pkg/occlient/occlient_test.go (hunk 3 above)
Comment (syamgk) [reply pkg/occlient/occlient_test.go]: sorry this is a WIP and i still have to add checks for the size of createdPVC

File: pkg/occlient/occlient_test.go (hunk 1 above)
Comment (syamgk) [reply pkg/occlient/occlient_test.go]: done

File: pkg/occlient/occlient_test.go [hunk 4]
```diff
@@ -8,9 +8,74 @@ import (
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fkclient, fkclientset := FakeNew()
+
+			_, err := fkclient.CreatePVC(tt.name, tt.size, tt.labels)
+
+			// Checks for error in positive cases
+			if !tt.wantErr == (err != nil) {
+				t.Errorf(" client.CreatePVC(name, size, labels) unexpected error %v, wantErr %v", err, tt.wantErr)
+				return
```
Comment (kadel) [inline pkg/occlient/occlient_test.go]: no need to call return here.

File: pkg/occlient/occlient_test.go [hunk 5]
```diff
@@ -8,9 +8,74 @@ import (
...
+			}
+
+			// Check for validating actions performed
+			if len(fkclientset.Kubernetes.Actions()) != 1 {
+				t.Errorf("expected 1 action in CreatePVC got: %v", fkclientset.RouteClientset.Actions())
+			}
+			// Checks for return values in positive cases
+			if err == nil {
+				createdPVC := fkclientset.Kubernetes.Actions()[0].(ktesting.CreateAction).GetObject().(*corev1.PersistentVolumeClaim)
+				quantity, _ := resource.ParseQuantity(tt.size)
```
Comment (kadel) [inline pkg/occlient/occlient_test.go]: errors shouldn't be ignored

File: pkg/occlient/occlient_test.go [hunk 6]
```diff
@@ -8,9 +8,74 @@ import (
...
 )
 
+func TestCreatePVC(t *testing.T) {
+	tests := []struct {
+		name    string
+		size    string
+		labels  map[string]string
+		wantErr bool
+	}{
+		{
```
Comment (kadel) [inline pkg/occlient/occlient_test.go]: there should also be some tests that result in error. For example invalid `size`  value

**Pull request merged**